src/layers/normalise.jl

```diff
@@ -0,0 +1,65 @@
+"""
+    ChannelDiag(α, β)
+    ChannelDiag(size::Integer)
+
+Create an element-wise linear layer, which performs
+
+    y = W .* x .+ b
+
+W and b are reshaped and broadcasted versions of
+α and β to match sizes along the channel dimension.
+
+The learnable arrays are initialised `α = ones(Float32, size)` and
+`β = zeros(Float32, size)`.
+
+Used by [`ChannelLayerNorm`](@ref).
+"""
+struct ChannelDiag{T}
+  α::T
+  β::T
+end
+
+function ChannelDiag(sz)
+  α = Flux.ones32(sz...)
+  β = Flux.ones32(sz...)
+  return ChannelDiag(α, β)
+end
+
+@functor ChannelDiag
+
+function (a::ChannelDiag)(x)
+  W = Flux.unsqueeze(Flux.unsqueeze(a.α, 1), 1)
+  b = Flux.unsqueeze(Flux.unsqueeze(a.β, 1), 1)
+  return W .* x .+ b
+end
+
+"""
+    ChannelLayerNorm(sz, λ = identity; affine = true, ϵ = 1f-5)
+
+A variant of LayerNorm where the input is normalised along the
+channel dimension. The input is expected to have channel 
+dimension with size `sz`.
+
+If `affine = true` also applies a learnable shift and rescaling
+as in the [`ChannelDiag`](@ref) layer.
+"""
+struct ChannelLayerNorm{F,D,T,N}
+  λ::F
+  diag::D
+  ϵ::T
+  size::NTuple{N,Int}
+  affine::Bool
+end
+
+function(m::ChannelLayerNorm)(x)
+  x = MLUtils.normalise(x, dims = ndims(x) - 1, ϵ = m.ϵ)
```

Comment:
Will do 👍🏽

Reply:
On some reflection and experimentation I've come to realise that this is harder than I first thought, mostly because of dimension generalisations. In this particular case it's easy enough to pass the right constructor to `Diagonal`, but in case of random dimension choices ensuring the broadcasts work out and that the dimensions are maintained as is might be a problem. There's a very user-specific option which involves exposing the entire API to the user and asking them to decide the dimension choices and ensure that the size inputs to the `Diagonal` match, but I'm not sure that's the way to go for Flux...

Reply:
Yes, you're right including the `(1, 1, ...)` style changes will get hairy. I guess if we want this to be generic, `LayerNorm` should operate on a contiguous set of dimensions with a leading offset. We normalize the `offset:offset+length(sz)` dimensions of `x`, and we pad `ntuple(() -> 1, offset)` to the `Diagonal`. I don't think it is possible to make the broadcasting work out right for non-contiguous dimensions.

This seems like a complex change that would require a nicer user-facing API design too. If you want to tackle it as well, that would be great @theabhirath. But for this PR, I am fine with the current level of code duplication.

Reply:
Yeah this is something I would definitely want to take up but I don't want to block this PR for that...I'll work on it separately and open a PR in Flux when I think I have something that can work well for this purpose